validation/validation_util.jl

```diff
@@ -42,3 +42,24 @@ function extract_number_from_filename(filename)
     end
     return -1
 end
+
+function extract_resolution_from_filename(str)
+    str = match(r"\d+(?!.*\d)", str).match
+
+    # Remove leading zeros and count them
+    leading_zeros = length(match(r"^0*", str).match)
+    str_non_zero = replace(str, r"^0*" => "")
+
+    if isempty(str_non_zero)
+        return "0.0"
+    end
+
+    # Adjust string to have a decimal point at the correct position
+    if leading_zeros > 0
+        decimal_str = "0." * "0"^(leading_zeros - 1) * str_non_zero
+    else
+        decimal_str = str_non_zero
+    end
+
+    return string(parse(Float64, decimal_str))
```

Comment:
```suggestion
    return decimal_str
```
Shouldn't this be the same?

Reply:
This handles a corner case I ran into e.g.:
julia> decimal_str = "3"
"3"

julia> string(parse(Float64, decimal_str))
"3.0"

But did do this differently before.

Reply:
Then please either remove it or add a comment explaining why this is necessary. Otherwise, it looks redundant.